starport/services/scaffolder/type.go

```diff
@@ -13,45 +12,97 @@ import (
 	"github.com/tendermint/starport/starport/pkg/xgenny"
 	modulecreate "github.com/tendermint/starport/starport/templates/module/create"
 	"github.com/tendermint/starport/starport/templates/typed"
+	"github.com/tendermint/starport/starport/templates/typed/basic"
 	"github.com/tendermint/starport/starport/templates/typed/indexed"
 	"github.com/tendermint/starport/starport/templates/typed/singleton"
 )
 
-type TypeModel string
+// AddTypeOption configures options for AddType.
+type AddTypeOption func(*addTypeOptions)
 
-const (
-	List      TypeModel = "list"
-	Map       TypeModel = "map"
-	Singleton TypeModel = "singleton"
-)
+type addTypeOptions struct {
+	moduleName string
+	fields     []string
+
+	isList      bool
+	isMap       bool
+	isSingleton bool
+
+	index string
+
+	withoutMessage bool
+}
+
+// ListType enables adding a list storage.
```

Comment:
I would say something like "allows to generate the methods to store the type instances as a list inside store"

And a similar format for the three others

Reply:
solved by 9204e56.